lib/ap/plan9/malloc.c

```diff
@@ -74,7 +74,7 @@ malloc(size_t size)
 	if(pow < CUTOFF) {
 		n = (CUTOFF-pow)+2;
 		bp = sbrk(size*n);
-		if((intptr_t)bp == -1)
+		if((long)bp == -1)
```

Comment:
That's going to bite you in the future, in fact the first time the bss grows past 2G this will break. Are you really sure you want to do this? It just looks wrong to me. Better to get the code into the 21st century :-)